src/sys/socket/mod.rs

```diff
@@ -48,18 +48,27 @@ pub use self::multicast::{
     ip_mreq,
     ipv6_mreq,
 };
-pub use self::consts::*;
 
 pub use libc::sockaddr_storage;
 
 #[derive(Clone, Copy, PartialEq, Eq, Debug)]
 #[repr(i32)]
 pub enum SockType {
-    Stream = consts::SOCK_STREAM,
-    Datagram = consts::SOCK_DGRAM,
-    SeqPacket = consts::SOCK_SEQPACKET,
-    Raw = consts::SOCK_RAW,
-    Rdm = consts::SOCK_RDM,
+    Stream = libc::SOCK_STREAM,
+    Datagram = libc::SOCK_DGRAM,
+    SeqPacket = libc::SOCK_SEQPACKET,
+    Raw = libc::SOCK_RAW,
+    Rdm = libc::SOCK_RDM,
+}
+
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[repr(i32)]
+pub enum SockProtocol {
+    Default = 0,
+    Tcp = libc::IPPROTO_TCP,
+    Udp = libc::IPPROTO_UDP,
+    #[cfg(any(target_os = "macos", target_os = "ios"))]
+    SysControl = libc::SYSPROTO_CONTROL,
 }
 
 // Extra flags - Supported by Linux 2.6.27, normalized on other platforms
```

Comment:
Can you replace this with `libc_bitflags!` as well? These constants should be available already in `libc` so this shouldn't be a problem. This would close #244.